### PR TITLE
Add gradle tasks to package generated platform sources

### PR DESCRIPTION
**Pull request by shardulm94**
- Added a `Packaging` class which represents configuration of how the generated source set will be packaged into a publishable artifact.
- Implemented two types of packaging
  - `DistributionPackaging` which creates TARs and ZIPs using the `DistributionPlugin`
  - `ShadedJarPackaging` which create Shaded JARs using the `ShadePlugin`
    - Since the Shade plugin is not open source yet, I have copied `ShadeTask` from the plugin with minor modifications as noted in the Javadoc. This can be referenced directly from the Shade plugin once it is available.

### CODE REVIEW

File: transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/packaging/Packaging.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import java.util.List;
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.TaskProvider;
+
+
+/**
+ * Creates and configures packaging tasks for a given platform which generate publishable/distributable artifacts.
+ * Usually this involves identify all dependencies to be included in the artifact, configuring tasks to generate these
```
Comment (wmoustafa) [inline transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/packaging/Packaging.java]: identifying

File: transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/packaging/ShadedJarPackaging.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import org.gradle.api.tasks.bundling.Jar;
+
+import static com.linkedin.transport.plugin.ConfigurationType.*;
+import static com.linkedin.transport.plugin.SourceSetUtils.*;
+
+
+/**
+ * A {@link Packaging} class which generates Shaded JARs containing all runtime dependencies using the
+ * {@link ShadeTask}
+ * TODO: Move to using the Shade Plugin once it is open-sourced
```
Comment (wmoustafa) [inline transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/packaging/ShadedJarPackaging.java]: Remove.

File: transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/tasks/ShadeTask.java [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.file.FileCollection;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.TaskAction;
+
+
+/**
+ * This code is borrowed from the Shade plugin which has not been open-sourced yet. Once the plugin is open-sourced, we
```
Comment (wmoustafa) [inline transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/tasks/ShadeTask.java]: Remove and provide proper java doc.

File: transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/packaging/ShadedJarPackaging.java (hunk 2 above)
Comment (rdsr) [reply transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/packaging/ShadedJarPackaging.java]: We should keep this here or I'm ok with a separate issue also

File: transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/tasks/ShadeTask.java (hunk 3 above)
Comment (rdsr) [reply transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/tasks/ShadeTask.java]: I'm negative to remove but I'm ok with a separate github issue created

File: transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/packaging/Packaging.java (hunk 1 above)
Comment (shardulm94) [reply transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/packaging/Packaging.java]: Fixed

File: transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/packaging/ShadedJarPackaging.java (hunk 2 above)
Comment (shardulm94) [reply transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/packaging/ShadedJarPackaging.java]: Removed TODO

File: transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/tasks/ShadeTask.java (hunk 3 above)
Comment (shardulm94) [reply transportable-udfs-plugin/src/main/java/com/linkedin/transport/plugin/tasks/ShadeTask.java]: Added java doc

**Pull request merged**